homeassistant/components/webostv/__init__.py

```diff
@@ -129,21 +133,58 @@ async def async_setup(hass, config):
         return True
 
     config_file = hass.config.path(WEBOSTV_CONFIG_FILE)
-    client_keys = await hass.async_add_executor_job(read_client_keys, config_file)
+    if not (
+        client_keys := await hass.async_add_executor_job(read_client_keys, config_file)
+    ):
+        _LOGGER.debug("No pairing keys, Not importing webOS Smart TV YAML config")
+        return True
+
+    async def async_migrate_task(entity_id, conf, key):
+        _LOGGER.debug("Migrating webOS Smart TV entity %s unique_id", entity_id)
+        client = WebOsClient(conf[CONF_HOST], key)
+        while not client.is_connected():
+            try:
+                await client.connect()
+            except WEBOSTV_EXCEPTIONS:
+                await asyncio.sleep(IMPORT_RECONNECT_DELAY)
```

Comment:
Maybe double the delay every iteration to backoff with a max of 1-2 minutes? It could be a while until the TV is turned on.


Example:
https://github.com/home-assistant/core/blob/5a33610739a20ab25121ecede4e76ac5a14ad892/homeassistant/config_entries.py#L353

Reply:
I like this backoff